doc/developer/demos.md

```diff
@@ -0,0 +1,29 @@
+# Developer guide: creating demos
```

Comment:
nit: the "developer guide" is meant to be a collection of documents that _could_ be read from start to finish by following the sections in guide.md. If this should be part of the developer guide, name it `guide-demos.md` and link it from guide.md, and stick it in the table of contents in the right spot. If not, then still add it to the table of contents, but not under the "developer guide" bullet, and drop the "Developer guide: " prefix from the heading!

Reply:
thanks for pointing this out! I'm thinking this doesn't belong in the developer guide (so I did the latter), but I'm happy to be overruled.

Reply:
I have the same weak preference!